cmd_watch.go

```diff
@@ -210,39 +221,46 @@ func waitCircle(parent context.Context, cfg watchConfig) (passed bool, started,
 // and decides whether the build has finished and if finished, whether it
 // failed. If an error is returned, it represents an error talking to the
 // CircleCI API, not an error with the workflow.
-func evalWorkflow(client *circleci.Client, wfID string, jobName string) (finished bool, failed bool, err error) {
+func evalWorkflow(client *circleci.Client, wfID string, jobName string) (anyRunning bool, anyFailed bool, anyBlocked bool, err error) {
 	fmt.Printf("%s: polling for jobs: ", time.Now().Format(time.StampMilli))
 	wfJobs, err := getJobs(client, wfID)
 	if err != nil {
 		fmt.Printf("error polling: %s\n", err.Error())
-		return true, true, err
+		return true, true, false, err
 	}
 	fmt.Println(summarizeJobList(wfJobs))
 
+	anyRunning = false
+	anyBlocked = false
 	for _, job := range wfJobs {
-		// always count ourself as finished so we don't wait for ourself
+		// skip ourself so we don't wait if we're the only job running
 		if job.Name == jobName {
 			continue
 		}
 
 		switch job.Status {
```

Comment:
should we introduce a `default` stanza here to deal with unknown CircleCI states in case they introduce a new one?

Reply:
yeah good call.  hm, what should that stanza actually do though?  ideally we'd include it in instrumentation, so we'd need to bubble it up someplace where we can send it.

Reply:
treat it as a failure state, I'd say?

Reply:
that worries me, anyone using circleci + buildevents would see failed builds if the watch job turns red, as opposed to broken traces

Reply:
maybe treat it such that we don't early exit and run through the 2 minute "we think we might be done, sleeping and trying again" loop in the hopes that it'll change to a state we recognize, and also send it up in instrumentation?  we can run a trigger against our own builds that would surface it.

Reply:
going to address this in another PR so we have more space/time to think about it 👍